### PR TITLE
Installation: Combined database errors

### PR DESCRIPTION
**Pull request by manishbisht**
Fixed the junior issue :  https://github.com/owncloud/core/issues/9465
First pull request to owncloud :)

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
requires jenkins pr

**Comment by manishbisht** [comment]
@DeepDiver1975 I am a student so what I have to write on the "Your Employer" field on the Contributor Agreement form ?

**Comment by DeepDiver1975** [comment]
> I am a student so what I have to write on the "Your Employer" field on the Contributor Agreement form ?

@karlitschek mind answering this? THX

**Comment by manishbisht** [comment]
And what is meant by "requires jenkins pr" can you please explain as i am new to this community.

**Comment by karlitschek** [comment]
@manishbisht You can leave the employer field open if it doesn't apply.

**Comment by MorrisJobke** [comment]
> @DeepDiver1975 What is meant by "requires jenkins pr" can you please explain as i am new to this community.

Currently some of our CI (jenkins in this case) jobs only work if the PR is based on a internal branch and not a branch of a fork. Therefore we create a branch based on your commits, push it into the core repo and create another PR (the jenkins pr) where the tests could run properly.

File: lib/private/setup/oci.php [hunk 1]
```diff
@@ -53,9 +53,12 @@ public function validate($config) {
 		if(empty($config['dbuser'])) {
 			$errors[] = $this->trans->t("%s enter the database username.", array($this->dbprettyname));
 		}
-		if(empty($config['dbname'])) {
+		else if(empty($config['dbname'])) {
 			$errors[] = $this->trans->t("%s enter the database name.", array($this->dbprettyname));
 		}
+		else if(empty($config['dbuser']&&empty($config['dbname'])) {
```
Comment (MorrisJobke) [inline lib/private/setup/oci.php]: This will never the reached, because one of the conditions before will met and then the else branch is never executed.

Comment (manishbisht) [inline lib/private/setup/oci.php]: Ohh !! Silly mistake done by me.
Rechecking the code and making some suggestions.
Will update you in 2-3 hours :)

**Comment by manishbisht** [comment]
I have updated the code and emailed the contributor agreement also. Please check @DeepDiver1975 @MorrisJobke @karlitschek :)

**Comment by karlitschek** [comment]
confirmed :-)

**Comment by manishbisht** [comment]
Excited to merge first pull request @karlitschek Got the tag of owncloud in my profile ;)

**Comment by manishbisht** [comment]
@karlitschek @DeepDiver1975 @MorrisJobke I have updated the source code and completed this junior job. Who will review this PR and how to merge this PR on the owncloud core ?

File: lib/private/setup/oci.php [hunk 2]
```diff
@@ -50,10 +50,13 @@ public function initialize($config) {
...
 		$errors = array();
-		if(empty($config['dbuser'])) {
+		if(empty($config['dbuser'])&&empty($config['dbname'])) {
+			$errors[] = $this->trans->t("%s enter the database username and name.", array($this->dbprettyname));
+		}
+		else if(empty($config['dbuser'])) {
 			$errors[] = $this->trans->t("%s enter the database username.", array($this->dbprettyname));
 		}
-		if(empty($config['dbname'])) {
+		else if(empty($config['dbname'])) {
```
Comment (MorrisJobke) [inline lib/private/setup/oci.php]: `} else if() {`

File: lib/private/setup/oci.php [hunk 3]
```diff
@@ -50,10 +50,13 @@ public function initialize($config) {
 
 	public function validate($config) {
 		$errors = array();
-		if(empty($config['dbuser'])) {
+		if(empty($config['dbuser'])&&empty($config['dbname'])) {
```
Comment (MorrisJobke) [inline lib/private/setup/oci.php]: space before and after the `&&`

File: lib/private/setup/oci.php [hunk 4]
```diff
@@ -50,10 +50,13 @@ public function initialize($config) {
...
+			$errors[] = $this->trans->t("%s enter the database username and name.", array($this->dbprettyname));
+		}
+		else if(empty($config['dbuser'])) {
```
Comment (MorrisJobke) [inline lib/private/setup/oci.php]: `} else if(){`

**Comment by manishbisht** [comment]
@MorrisJobke Changes completed. :)

**Comment by manishbisht** [comment]
@karlitschek @DeepDiver1975 @MorrisJobke Please Review this PR.

**Comment by MorrisJobke** [comment]
@manishbisht I just tested it and could you please change the code at https://github.com/owncloud/core/blob/08f0dc906780be699aa9ff8c10faaea0be44855d/lib/private/setup/abstractdatabase.php#L62-L67 in the same way? Then I'm fine with this :)

**Comment by manishbisht** [comment]
@karlitschek @DeepDiver1975 @MorrisJobke Made the changes on the https://github.com/owncloud/core/blob/08f0dc906780be699aa9ff8c10faaea0be44855d/lib/private/setup/abstractdatabase.php#L62-L67. Please review. :)

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by MorrisJobke** [comment]
Let me create a internal branch, because CI only works properly based on a core branch.

**Comment by MorrisJobke** [comment]
> Let me create a internal branch, because CI only works properly based on a core branch.

CI PR: https://github.com/owncloud/core/pull/23326

**Comment by DeepDiver1975** [comment]
https://github.com/owncloud/core/pull/23326 was merged - @manishbisht Thanks a lot and welcome to the show! :wink:

**Pull request closed without merging**